cmd/jaas/cmd/registercontroller.go

```diff
@@ -66,59 +91,105 @@ func (c *registerControllerCommand) SetFlags(f *gnuflag.FlagSet) {
 		"json": cmd.FormatJson,
 	})
 	c.file.StdinMarkers = stdinMarkers
+	f.BoolVar(&c.local, "local", false, "If local flag is specified, then the local API address and CA cert of the controller will be used.")
+	f.BoolVar(&c.dryRun, "dry-run", false, "Dry-run enabled will only print the controller details.")
+	f.StringVar(&c.tlsHostname, "tls-hostname", "", "Specify the hostname for TLS verification.")
+	f.StringVar(&c.file.Path, "file", "", "Specify a file-path for controller details, use '-' to read from stdin.")
 }
 
 // Init implements the cmd.Command interface.
 func (c *registerControllerCommand) Init(args []string) error {
 	if len(args) < 1 {
-		return errors.E("filename not specified")
+		return errors.New("controller name not specified")
 	}
-	c.file.Path = args[0]
+	c.controllerName = args[0]
 	if len(args) > 1 {
-		return errors.E("too many args")
+		return errors.New("too many args")
 	}
 	return nil
 }
 
 // Run implements Command.Run.
 func (c *registerControllerCommand) Run(ctxt *cmd.Context) error {
+	data, err := c.getControllerDetails(ctxt)
+	if err != nil {
+		return err
+	}
+	var params apiparams.AddControllerRequest
+	if err = unmarshalControllerDetails(&params, data); err != nil {
+		return err
+	}
+	if c.controllerName != params.Name {
+		return errors.New(fmt.Sprintf("provided controller name doesn't match, %s != %s", c.controllerName, params.Name))
+	}
+	if c.dryRun {
+		_ = c.out.Write(ctxt, params)
```

Comment:
```suggestion
		return c.out.Write(ctxt, params)
```

Reply:
That would print the error if the write failed, which I guess is a good thing? So will change :+1: